# <center><font color='blue'>IS362 Assignment 5 Fall 2020</center><font>

![My_Text](https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcSZtgSr9JkT6mwn474na4hRPHouehtMwM94lQ&usqp=CAU) 

## Our task for this week's  assignment is to answer three questions:
 1. What is the northern most airport in the United States?
 2. What is the eastern most airport in the United States?
 3. On February 12th, 2013, which New York area airport had the windiest weather?

### Our Tasks

#### 1. Use the information in the .CSV files found at https://github.com/hadley/nycflights13/tree/master/data-raw for your source data.
#### 2. Write your code using Python and pandas, and deliver it in a Jupyter notebook. You should post your notebook in a GitHub repository, and send me a link to the notebook.
#### 3. For each of the three questions listed above, you’ll need to verify your answers (in two cases by searching for additional confirming information, and in the third case being alert for possible outliers). For the first two questions above, you may want to start by generating a list of the five northernmost or easternmost U.S. airports.
#### 4. In addition to providing code, you should state your answers to each of the above questions (in Markdown) in the Jupyter notebook.
#### 5. To receive full credit, you’ll need to include in your Jupyter notebook both your code and your documented assumptions and any changes that you made to the data based on your research (using Markdown) in your Jupyter notebook.

# <center><font color='blue'>Question 1.  What is the northernmost airport in the United States?</center><font>

In [40]:
import pandas as pd
import math
import datetime
import urllib.request
import json
northernmost_airport_in_us = pd.read_csv(r'C:\Users\MATTHEWS-PC\Desktop\IS362_Week5\data-raw\airports.csv')
northernmost_airport_in_us.head()
northernmost_airport_in_us.sort_values("lat", ascending = False).head(5)

faa                                        name        lat         lon  \
417  EEN                     Dillant Hopkins Airport  72.270833   42.898333   
230  BRW                  Wiley Post Will Rogers Mem  71.285446 -156.766003   
110  AIN                          Wainwright Airport  70.638056 -159.994722   
708  K03                               Wainwright As  70.613378 -159.860350   
152  ATK  Atqasuk Edward Burnell Sr Memorial Airport  70.467300 -157.436000   

     alt  tz dst              tzone  
417  149  -5   A                NaN  
230   44  -9   A  America/Anchorage  
110   41  -9   A  America/Anchorage  
708   35  -9   A  America/Anchorage  
152   96  -9   A  America/Anchorage

### <center><font color='blue'>The northern most airport in the US based on the dataset provided is Dillant Hopkins Airport.</center><font>

The definition of latitude is a a geographic coordinate that is very specific about the north-south position of a point
on the Earth's surface. It can be used to measuring the distance away from the equator.The Lat number will increase in 
size the further away we move from the equator.Dillant Hopkins Airport has a latitude of 72.270833 which is the airport
that's the furthest away from the equator.Which marks Dillant Hopkins Airport as the Northernmost Airport based on the information given

https://www.google.com/maps/dir/5910+Junction+Blvd,+Queens,+NY+11373,+USA/Key+West,+Florida/@33.9633109,-85.6008169,5.21z/data=!4m14!4m13!1m5!1m1!1s0x89c25e358b8e2823:0x7a0658150f4d4b5d!2m2!1d-73.8652519!2d40.7347381!1m5!1m1!1s0x88d1b134ad952377:0x3fcee92f77463b5e!2m2!1d-81.7799871!2d24.5550593!3e1?hl=en&authuser=0

![img](https://i.imgur.com/dPnctLW.jpeg)
<br>
            


# <center><font color='blue'>Question 2.What is the easternmost airport in the United States?</center><font>


In [24]:
easternmost_airport_in_us = pd.read_csv(r'C:\Users\MATTHEWS-PC\Desktop\IS362_Week5\data-raw\airports.csv')
eastern = eastern_airport_in_us
easternmost_airport_in_us = eastern[(eastern['lat'] > 44) & (eastern['lat'] <= 52)]
easternmost_airport_in_us.head()
easternmost_airport_in_us.sort_values("lon", ascending = False).head(5)

faa                                 name        lat        lon  alt  tz  \
444   EPM           Eastport Municipal Airport  44.910111 -67.012694   45  -5   
624   HUL                         Houlton Intl  46.123083 -67.792056  489  -5   
259   CAR                         Caribou Muni  46.871500 -68.017917  626  -5   
1101  PQI  Northern Maine Rgnl At Presque Isle  46.688958 -68.044797  534  -5   
1398  WFK  Northern Aroostook Regional Airport  47.285556 -68.312778  988  -5   

     dst             tzone  
444    A  America/New_York  
624    A  America/New_York  
259    A  America/New_York  
1101   A  America/New_York  
1398   A  America/New_York

### <center><font color='blue'>The eastern most airport in the US based on the dataset provided is Eastport Municipal Airport.</center><font>
Longitude lines can be described as the measurement from east to west of th eprime meridian.The easatern most airport being the
Eastport Municipal Airport with a longitude of -67.012694 which is the most eastern location on the .csv given.

https://www.google.com/maps/place/Eastport+Municipal+Airport-EPM/@42.7042733,-64.2776661,5.9z/data=!4m5!3m4!1s0x4ca8bcbb416bb557:0xe8e7c4805bb5f1e8!8m2!3d44.9099796!4d-67.0119367

![img](https://i.imgur.com/Uar7xfV.jpg)
<br>

# <center><font color='blue'>Question 3.  On February 12th, 2013, which New York area airport had the windiest weather?</center><font>

In [51]:
weather = pd.read_csv(r'C:\Users\MATTHEWS-PC\Desktop\IS362_Week5\data-raw\weather.csv')
windiest_weather_ny = weather[(weather["month"]==2) & (weather["day"]==12) & (weather["year"]==2013)]
windiest_weather_ny.head()
windiest_weather_ny.sort_values("wind_speed", ascending=False).head(5)

origin  year  month  day  hour   temp   dewp  humid  wind_dir  \
1009     EWR  2013      2   12     3  39.02  26.96  61.63     260.0   
18417    LGA  2013      2   12     2  42.98  26.06  50.94     290.0   
1018     EWR  2013      2   12    12  44.06  26.06  48.87     270.0   
18428    LGA  2013      2   12    13  44.06  23.00  43.02     300.0   
18429    LGA  2013      2   12    14  44.06  23.00  43.02     300.0   

       wind_speed  wind_gust  precip  pressure  visib             time_hour  
1009   1048.36058        NaN     0.0    1008.3   10.0  2013-02-12T08:00:00Z  
18417    23.01560   31.07106     0.0    1007.1   10.0  2013-02-12T07:00:00Z  
1018     21.86482   31.07106     0.0    1012.5   10.0  2013-02-12T17:00:00Z  
18428    21.86482   25.31716     0.0    1011.7   10.0  2013-02-12T18:00:00Z  
18429    20.71404   25.31716     0.0    1011.5   10.0  2013-02-12T19:00:00Z

### <center><font color='blue'>The New York area airport that had the windiest weather on February 12th, 2013 would be EWR based on the dataset provided.</center><font>

Based on the data EWR was the windiest aiport on frebruary 12th 2013